bonsai-core/src/main/scala/com/stripe/bonsai/IndexedBitSet.scala

```diff
@@ -160,20 +244,53 @@ object IndexedBitSet {
     }
 
   /**
-   * TODO: maybe tom can explain this?
+   * Returns the rank of the i-th bit in a 32-bit word.
    */
-  def rankWord(word: Int, i: Int): Int = {
+  private[bonsai] def rankWord(word: Int, i: Int): Int = {
     val mask = ~(-1L << ((i + 1))).toInt
     java.lang.Integer.bitCount(word & mask)
   }
 
   /**
+   * Returns the index of the i-th set bit in a 32-bit word.
+   *
+   * @param word a 32-bit word to search
+   * @param rank a value in [1,32] (inclusive)
+   */
+  private[bonsai] def selectWord(word: Int, rank: Int): Int = {
```

Comment:
So, this is essentially doing a binary search on `word`, looking for the `rank`-th set bit. What I'm not super happy with is the number of `bitCounts` used. I mean, this will probably be fast relative to whatever is using this, just because it's all operating on a handful of stack-allocated integers that'll be in the cache, but `bitCount` is a bit involved by itself. Still, I feel like there is probably a better way to do this, but my trusty copy of Hackers Delight doesn't seem to have this exact thing.